courier/test/persistence.go

```diff
@@ -107,6 +107,32 @@ func TestPersister(ctx context.Context, newNetworkUnlessExisting NetworkWrapper,
 			assert.Equal(t, originalSendCount+1, ms[0].SendCount)
 		})
 
+		t.Run("case=list messages", func(t *testing.T) {
+			status := courier.MessageStatusProcessing
+			filter := courier.MessagesFilter{
+				Status: &status,
+				PaginationParams: x.PaginationParams{
+					Page:    1,
+					PerPage: 100,
+				},
+			}
+			ms, tc, err := p.ListMessages(ctx, filter)
+
+			require.NoError(t, err)
+			assert.Len(t, ms, len(messages))
+			assert.Equal(t, int64(len(messages)), tc)
+			assert.Equal(t, messages[len(messages)-1].ID, ms[0].ID)
```

Comment:
```suggestion
			assert.Equal(t, messages[len(messages)-1].ID, ms[0].ID)
			assert.ElementsMatch(t, messages, ms)
```
`assert.ElementsMatch` ignores the order.`

Reply:
Didn't know that, thank you :-)

Reply:
@zepatrik On a second thought, this can't work on this scenario. Test cases above the `case=list messages` either updates the messages with `SetMessageStatus` or `IncrementMessageSendCount` which none of them takes a reference to the `courier.Message` itself. So `ListMessages` method always returns messages with different `MessageStatus` or `SendCount` than what we have in `messages` variable. So `assert.ElementsMatch` statement bound to be failed due to the reasons i listed above.